### ISSUE TITLE
[jit] `random` module support

### ISSUE THREAD

**Issue by driazati** [pytorch/pytorch#25803]
This is used in MaskRCNN (specifically `random.choice`), we could implement the subset of `random` that's not covered in `torch` already

cc @suo

**Comment by urandu** [comment]
I would like to tackle this issue as my first contribution, is that okay? @driazati

**Comment by driazati** [comment]
Yeah sure, I would start by just implementing `random.choice` (and adding a test). 

Some relevant code pointers:

* https://github.com/pytorch/pytorch/blob/ee312b134ca6772099ecfff1ebab068cc53261e4/torch/jit/__init__.py#L2005
* the file `torch/csrc/jit/register_prim_ops.cpp`

**Comment by eellison** [comment]
I think we want to be careful about implementing random. We won't be using the same RNG as `random` is, so we will be silently changing the reproducibility of the model.

**Comment by urandu** [comment]
@eellison which approach do you suggest?

**Comment by eellison** [comment]
@urandu I'm not sure. I don't think we should add it for now, especially without an actual user asking for it (@driazati is a pytorch dev). 

Maybe try https://github.com/pytorch/pytorch/issues/25804 or another `good-first-issue` instead?

**Comment by urandu** [comment]
Thanks @eellison  I will try out #25804  instead.

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
